pkg/cache/v3/eventcounter.go

```diff
@@ -106,6 +153,40 @@ func getUserValues(cmds []*goredis.IntCmd) ([]float64, error) {
 	return userVals, nil
 }
 
+func (c *eventCounterCache) GetUserCountsV2(keys [][]string) ([]float64, error) {
+	pipe := c.cache.Pipeline()
+	intCmds := make([][]*goredis.IntCmd, 0, len(keys))
+	for _, day := range keys {
+		hourlyCmds := []*goredis.IntCmd{}
+		for _, hour := range day {
+			c := pipe.PFCount(hour)
+			hourlyCmds = append(hourlyCmds, c)
+		}
+		intCmds = append(intCmds, hourlyCmds)
+	}
+	_, err := pipe.Exec()
+	if err != nil {
+		return []float64{}, fmt.Errorf("err: %v, keys: %v", err, keys)
+	}
+	return getUserValuesV2(intCmds)
+}
+
+func getUserValuesV2(cmds [][]*goredis.IntCmd) ([]float64, error) {
+	userVals := make([]float64, 0, len(cmds))
+	for _, day := range cmds {
+		var totalVal float64
+		for _, hour := range day {
+			val, err := hour.Result()
+			if err != nil {
+				return []float64{}, err
+			}
+			totalVal += float64(val)
```

Comment:
Because the user count is unique, we can't just sum the values. Otherwise, it can report wrong values.
If the same user sends an event every hour during the day, the current implementation will count as 24 events, but it should be reported as 1.

We will need to use `PFMERGE` instead.
https://redis.io/commands/pfmerge/